src/tiered/tiered_opener.c

```diff
@@ -19,45 +19,48 @@ __tiered_opener_open(WT_BLOCK_FILE_OPENER *opener, WT_SESSION_IMPL *session, uin
     WT_BUCKET_STORAGE *bstorage;
     WT_DECL_RET;
     WT_TIERED *tiered;
+    u_int open_flags;
     const char *object_name, *object_uri;
+    bool local_only;
 
     tiered = opener->cookie;
     object_uri = NULL;
+    open_flags = flags;
+    local_only = false;
 
     WT_ASSERT(session,
       (object_id > 0 && object_id <= tiered->current_id) || object_id == WT_TIERED_CURRENT_ID);
     /*
-     * FIXME-WT-7590 we will need some kind of locking while we're looking at the tiered structure.
-     * This can be called at any time, because we are opening the objects lazily.
+     * First look for the local file. This will be the fastest access and we retain recent objects
+     * in the local database for a while.
      */
     if (object_id == tiered->current_id || object_id == WT_TIERED_CURRENT_ID) {
         bstorage = NULL;
         object_name = tiered->tiers[WT_TIERED_INDEX_LOCAL].name;
         if (!WT_PREFIX_SKIP(object_name, "file:"))
```

Comment:
I realize this isn't part of your change, but I'm wondering why we specifically return an error here if we don't have a `file:` uri, whereas a few lines later we use `WT_PREFIX_SKIP_REQUIRED()` which asserts if we don't have an `object:` uri.

Reply:
I changed it to your suggestion. That feels cleaner.